unstructured/metrics/text_extraction.py

```diff
@@ -82,3 +89,47 @@ def bag_of_words(text: str) -> Dict[str, int]:
                     bow[incorrect_word] = 1
             i = j
     return bow
+
+
+def calculate_percent_missing_text(
+    output: str,
+    source: str,
+) -> float:
+    """
+    Creates the bag of words (BOW) found in each input text and their frequencies, then compares the
+    output BOW against the source BOW to calculate the % of text from the source text missing from
+    the output text.
+
+    Takes "clean, concatenated text" (CCT) from a document output and the ground truth source text
+    as inputs.
+
+    If the output text contains all words from the source text and then some extra, result will be
+    0% missing text - this calculation does not penalize duplication.
+
+    A spaced-out word (ex. h e l l o) is considered missing; individual characters of a word
+    will not be counted as separate words.
+
+    Returns the percentage of missing text represented as a decimal between 0 and 1.
+    """
+    output_bow = bag_of_words(output)
+    source_bow = bag_of_words(source)
+
+    # get total words in source bow while counting missing words
+    total_source_word_count = 0
+    total_missing_word_count = 0
+
+    for source_word, source_count in source_bow.items():
+        total_source_word_count += source_count
+        if source_word not in output_bow:
+            # entire count is missing
+            total_missing_word_count += source_count
+        else:
+            output_count = output_bow[source_word]
+            total_missing_word_count += max(source_count - output_count, 0)
```

Comment:
Just to clarify, we are not penalizing any extra words extracted right?

Reply:
Correct, we are not penalizing duplicated text. If the unstructured output text has everything from the source text and then more, we count that is 0% missing text.